### PR TITLE
Release pipeline in GHA

### LINKED ISSUE #11: nri-ecs release binary/manifest
https://fsi-build.pdx.vm.datanerd.us/view/all/job/nri-ecs-release-binary

https://fsi-build.pdx.vm.datanerd.us/view/all/job/nri-ecs-release-manifests

Commands are already grouped in makefile targets that can be called

### PR DESCRIPTION
**Pull request by roobre**
Closes #11

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Upload tarball and configuration files to https://download.newrelic.com
+        run: |
+          make release_tarball_package_for OS=$OS ARCH=$ARCH RELEASE_VERSION=$RELEASE_VERSION
+          make upload_manifests RELEASE_VERSION=$RELEASE_VERSION INFRA_BUNDLE_VERSION=$INFRA_BUNDLE_VERSION
```
Comment (roobre) [inline .github/workflows/release.yml:46]: I am slightly concerned by `$INFRA_BUNDLE_VERSION` here. In the old J̶ȩ͏̨nk̛į̸n̸̕s̀͠  system, it was defaulted to `1.5.0.` but could be changed manually on each run.
Perhaps we should consider obtaining this from somewhere.

Comment (roobre) [reply .github/workflows/release.yml]: I have removed the override in 770ddf561b102211576e9b377cb5c9127c1922ce, this way the bundle version won't be referred in yet another place.

**Comment by roobre** [comment]
Rebased to include #15

**Review by paologallinaharbur — DISMISSED** [review]

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+      - name: Build tarball
+        run: |
+          make test
+          make package_for OS=$OS ARCH=$ARCH RELEASE_VERSION=$RELEASE_VERSION
+
+      - uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
```
Comment (roobre) [inline .github/workflows/release.yml:41]: Point to real secret names before merge

**Review by arvdias — DISMISSED** [review]

**Review by arvdias — APPROVED** [review]

**Review by paologallinaharbur — APPROVED** [review]

**Pull request merged**